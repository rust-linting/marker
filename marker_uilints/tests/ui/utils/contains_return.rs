#![allow(unused)]

fn test_contains_return_1_false() {
    // False, doesn't contain anything
}

fn test_contains_return_2_false() {
    if (16 << 2) < 2 {
        println!("IDK");
    } else {
        println!("idk");
    }
}

fn test_contains_return_3_true() {
    // True, is this code useful? Nope, but it contains a return
    return;
}

fn test_contains_return_4_true() -> Option<u32> {
    // True, is this code useful? Still nope, somehow it's worse
    let x: u32 = Some(2)?;
    Some(x)
}

fn test_contains_return_5_false() {
    // False, the return is nested
    fn nested_function() {
        return;
    }
    nested_function();
}
