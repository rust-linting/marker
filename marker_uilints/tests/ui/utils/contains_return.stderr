warning: testing `contains_return` -> false
 --> $DIR/contains_return.rs:3:4
  |
3 | fn test_contains_return_1_false() {
  |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(marker::test_contains_return)]` on by default

warning: testing `contains_return` -> false
 --> $DIR/contains_return.rs:7:4
  |
7 | fn test_contains_return_2_false() {
  |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: testing `contains_return` -> true
  --> $DIR/contains_return.rs:15:4
   |
15 | fn test_contains_return_3_true() {
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: testing `contains_return` -> true
  --> $DIR/contains_return.rs:20:4
   |
20 | fn test_contains_return_4_true() -> Option<u32> {
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: testing `contains_return` -> false
  --> $DIR/contains_return.rs:26:4
   |
26 | fn test_contains_return_5_false() {
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 5 warnings emitted

